### DISCUSSION TITLE
OIDC & LDAP Access Question

### DISCUSSION THREAD

**Discussion by heasterday** [quay/quay#1821] [Q&A] [answered]
Hello Quay Team!

Question for you all, if Quay is configured with LDAP and OIDC side by side, what happens when a user attempts to login via OIDC when they have been removed from the LDAP group? Does OIDC login just verify the user exists in the DB or is an additional ldap lookup performed to verify they still are in the backing LDAP group?

Let me know if you would like me to expand on the scenario above.
Thanks!

**Comment by michaelalang** [comment, marked as answer]
AFAIK, OIDC provides all necessary information to complete a login process and not `additional` lookup is done. Otherwise you could not use OIDC only. 

Which use-case are you looking for with the LDAP groups (Team,superuser?) 
OIDC does not utilize groups right now and superuser lists need to be maintained in config.yaml.
TeamSync is not implemented with OIDC as well and we've discussed the topic yesterday in the Quay Office hours call. It's on the roadmap but most likely not prior 3.10.
Typically, OIDC will provide the assigned groups with the JWT and the authorization backend needs to ensure upon login of the user, that it gets group/auth mappings done. This means, if people are not logging in, they would stay in groups until they login the next time (that is the workflow of OCP for example)

All the best and kind regards
Michi

**Reply by heasterday** [reply]
Thank you for the response! I was interested in using OIDC for AuthN and LDAP for AuthZ and looking into whether that is possible or not. I have the information I need so marking this answered.
